.github/workflows/annotate.yml

```diff
@@ -0,0 +1,55 @@
+name: Annotate checks
+
+on:
+  workflow_run:
+    workflows: ["ci"]
+    types:
+      - completed
+
+jobs:
+  annotate:
+    runs-on: ubuntu-latest
+    if: >
+      ${{ github.event.workflow_run.event == 'pull_request' }}
+    steps:
+      - name: 'Download artifact'
+        uses: actions/github-script@v5
+        with:
+          script: |
+            const fs = require('fs');
+            const opts = github.rest.actions.listWorkflowRunArtifacts.endpoint.merge({
+               owner: context.repo.owner,
+               repo: context.repo.repo,
+               run_id: ${{github.event.workflow_run.id }},
+            });
+            const artifacts = await github.paginate(opts);
+            for (const artifact of artifacts) {
+              if (!artifact.name.startsWith('test report ')) {
```

Comment:
This is an unwritten contract, should be mentioned somewhere.

Reply:
I added comments in the other workflow file.